src/vm/checker/typecheck/natives/mod.rs

```diff
@@ -173,8 +173,8 @@ fn check_special_let(checker: &mut TypeChecker, args: &[SymbolicExpression], con
 
         checker.contract_context.check_name_used(var_name)?;
 
-        if out_context.variable_types.contains_key(var_name) {
-            return Err(CheckError::new(CheckErrors::NameAlreadyUsed(var_name.to_string())))
+        if let Some(existing_var_name) = out_context.lookup_variable_type(var_name) {
+            return Err(CheckError::new(CheckErrors::NameAlreadyUsed(existing_var_name.to_string())))
```

Comment:
I think that this returns the string representation of the current variable's type. This would probably be better if it just checked `.is_some()` and still used `var_name`.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1036/commits/1d84f9b7c7baaa2fb62200d0de2e61417d98505a